### ISSUE TITLE
Errors - 2 ERROR  during startup - failed to initialize and failed to load this item

### ISSUE THREAD

**Issue by EchocallingEarth** [Slimefun/Slimefun4#3729]
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

Console errors on startup -- See ERROR 1 and ERROR 2 listed here
ERROR 1
[Slimefun] An Error occurred while initializing SlimefunItems for Slimefun DEV - 1035 (git 0b7c656d)
java.lang.IllegalArgumentException: Cannot send a warning for an unregistered item!
	at io.github.thebusybiscuit.slimefun4.libraries.commons.lang.Validate.notNull(Validate.java:192) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem.warn(SlimefunItem.java:1006) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.items.multiblocks.TableSaw.getPlanks(TableSaw.java:90) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.items.multiblocks.TableSaw.<init>(TableSaw.java:56) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.setup.SlimefunItemSetup.setup(SlimefunItemSetup.java:1035) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.Slimefun.loadItems(Slimefun.java:682) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.Slimefun.onPluginStart(Slimefun.java:324) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.Slimefun.onEnable(Slimefun.java:240) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[purpur-api-1.19.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:371) ~[purpur-api-1.19.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:548) ~[purpur-api-1.19.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_19_R1.CraftServer.enablePlugin(CraftServer.java:611) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at org.bukkit.craftbukkit.v1_19_R1.CraftServer.enablePlugins(CraftServer.java:525) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:641) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:427) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:343) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1116) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:310) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]

ERROR 2

[20:15:36] [Server thread/ERROR]: [Slimefun] Item "STAFF_ELEMENTAL_STORM" from Slimefun vDEV - 1035 (git 0b7c656d) has caused an Error!
[20:15:36] [Server thread/ERROR]: [Slimefun] You can report it here: https://github.com/Slimefun/Slimefun4/issues

[Server thread/ERROR]: [Slimefun] Failed to properly load this Item
java.lang.NullPointerException: Cannot invoke "io.github.thebusybiscuit.slimefun4.implementation.items.altar.AncientAltar.getRecipes()" because "altar" is null
	at io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType.lambda$static$0(RecipeType.java:54) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType.register(RecipeType.java:124) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem.load(SlimefunItem.java:787) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.setup.PostSetup.loadItems(PostSetup.java:73) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at io.github.thebusybiscuit.slimefun4.implementation.tasks.SlimefunStartupTask.run(SlimefunStartupTask.java:49) ~[Slimefun4 - DEV 1035 (git 0b7c6).jar:?]
	at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1130) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:310) ~[purpur-1.19.2.jar:git-Purpur-1857]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]

### 📑 Reproduction Steps

Errors during startup - Fresh install

### 💡 Expected Behavior

No errors

### 📷 Screenshots / Videos

NA

### 📜 Server Log

https://pastebin.com/mWtNPkbH

### 📂 `/error-reports/` folder

Nothing found in folder

### 💻 Server Software

Purpur

### 🎮 Minecraft Version

1.19.x

### ⭐ Slimefun version

Slimefun4 - DEV 1035 (git 0b7c6).jar

### 🧭 Other plugins

Plugins (81): AureliumSkills, Bank, BetterLogStrip, BetterTridents, BKCommonLib, BlockEffects, BlueMap, ccNoEnderpearlDamage*, ChatControlRed, ChestProtect, ChestSort, Citizens, ClearFog, CMI, CMILib, ColorBundles, ConsoleSay*, ConsoleSpamFix, Core, CraftBook, CustomCrafting, DecentHolograms, DisplayShops, DoorsReloaded, dtlTradersPlus, DyedBackpacks, EpicFarming, EpicHoppers, ExtraGear, ExtraTools, FastAsyncWorldEdit (WorldEdit), HeadDatabase, ImageFrame, InteractionVisualizer, InteractiveBoard, Item-Chest-Sorter, ItemsAdder, Jobs, LibsDisguises, LightCleaner, LoneLibs, Lootin, LuckPerms, MCPets, MobColors, ModelEngine, My_Worlds, MythicCrucible, MythicMobs, NamelessMC, NBTAPI, NetheriteShield, NoPlugins, ntdLuckyBlock, PerWorldInventory, PlaceholderAPI, PlotSquared, ProtocolLib, PublicCrafters, RealisticSeasons, RealisticVillagers, Redye, Residence, Sentinel, Slimefun, SnowierSnow, spark, Spartan, StackResize, SurvivalInvisiframes, TransportPipes, UltimateTimber, UpgradeableSpawners, VanillaTweaks, Vault, VirtualBooks, Wailat, WolfyUtilities, WorldBorder, WorldGuard, WorldGuardExtraFlags

**Comment by Boomer-1** [comment]
are there datapacks installed?

**Comment by EchocallingEarth** [comment]
Yes, we do use Datapacks - 

Please see the list below of packs in use with links if needed:

[ChoiceTheorem's Overhauled villages for 1.19](https://www.planetminecraft.com/data-pack/choicetheorem-s-overhauled-villages-for-1-19/)
[MC Augmentor | Sequoia Trees](https://www.planetminecraft.com/data-pack/mc-augmentor-sequoia-trees/)
[Incendium ~ Nether Expansion [Now 1.19!]](https://www.planetminecraft.com/data-pack/incendium-nether-expansion/)
[1.19 Improved Desert Village Structure Datapack](https://www.planetminecraft.com/data-pack/1-14-x-improved-desert-village-datapack/)
[Mo Villagers (v3.2)](https://www.planetminecraft.com/data-pack/mo-villagers-1-14-x/)
[WallsOfConquest_](https://www.planetminecraft.com/data-pack/wallsofconquest/)
[The Ancient City Overhaul v1.5](https://www.planetminecraft.com/data-pack/the-ancient-city-overhaul/)
[Mangrove Ruins](https://www.planetminecraft.com/data-pack/mangrove-ruins/)
Towns & Towers (1.18.2 and 1.19.2)
[Whimzee's Dungeon Overhaul](https://www.planetminecraft.com/data-pack/whimzee-s-dungeon-overhaul/)
[Dungeons + (Updates vanilla dungeons & adds new ones)](https://www.planetminecraft.com/data-pack/vanilla-dungeouns/)
[Cavernous [1.19.x]](https://www.planetminecraft.com/data-pack/thxlotl-s-cave-biome-datapack/)
[Fishing+](https://www.planetminecraft.com/data-pack/fishing-5772171/)
[Better Loot Tables (structures, mobs, blocks)!](https://www.planetminecraft.com/data-pack/better-loot-tables-mobs-structures-blocks/)
[Immersive Soundscapes](https://www.planetminecraft.com/data-pack/immersive-sounding/)
[Immersive Structures - Additional structures pack]([url](https://www.planetminecraft.com/data-pack/immersive-structure-additional-structures-pack/))
[Gald's Little Details | A collection of mini features](https://www.planetminecraft.com/data-pack/gald-s-little-details/)
[Towns & Towers (1.18.2 and 1.19.2)]([url](https://www.planetminecraft.com/data-pack/towns-amp-towers-structure-overhaul/))
[Splatus's Mountain Villages ~ [v1.0.2] [1.19]](https://www.planetminecraft.com/data-pack/splatus-s-mountain-villages-v1-0-1-18-2/)
[New Shipwrecks | Better Structures](https://www.planetminecraft.com/data-pack/new-shipwrecks-better-structures/)
[New End Cities | Better Structures](https://www.planetminecraft.com/data-pack/new-end-cities-better-structures/)
[New Ocean Ruins | Better Structures](https://www.planetminecraft.com/data-pack/new-ocean-ruins-better-structures/)
["1.19" Hopo Better Mineshafts "9 Types of Mineshafts"](https://www.planetminecraft.com/data-pack/better-mineshafts/)

**Comment by Boomer-1** [comment]
we don't need the links, but something on your list is causing the issue. If you can identify which one it is, it may help. Ultimately the issue needs to be reported with the datapack causing the issue.

**Comment by EchocallingEarth** [comment]
I will do a datapack 1 by 1 test till I find the culprit. And approach from there. Will this in theory fix both errors as reported?

**Comment by EchocallingEarth** [comment]
Also is there any info in what I have provided that may be helpful to the datapack creator once I narrow it down?

**Comment by Boomer-1** [comment]
if you remove the ones causing the issues yes. However, that won't fix the issue. that will simply let you know which ones to reach out to. The dev team here can look into it to see what may be causing the issue, and potentially decide where the fix has to be.

**Comment by EchocallingEarth** [comment]
Wonderful I will begin this process as soon as I get home from pucking up dinner. Hopefully will be fairly easy to find given I have a plan to approach the issue. Thank you I will update further later today.

**Comment by EchocallingEarth** [comment]
After some testing, I have discovered the Datapack that is causing the issue - 
It is [Gald's Little Details | A collection of mini features](https://www.planetminecraft.com/data-pack/gald-s-little-details/)

This leads me to inquire if this able to be approached from the side of Slimefun for a fix. And when bringing to the attention of the datapack creator is there anything I can tell them that would help with fixing the conflict?

**Comment by Boomer-1** [comment]
For now I would open an issue with them, provide them with the complete error log and everything you did to identify why you believe it was that datapack causing the error. It may be on them to fix, it may be on sf. the dev's can work together. I would link this to their report as well, so they can collab

**Comment by EchocallingEarth** [comment]
Opened a ticket or more accurately directed them to this post and will await their reply here as I am hopeful that they are open to working with the Slimefun team on a fix and that this fix possibly opens the datapack to more widely accepted use not only in general but in the Slimefun community as well.

**Comment by galdevear** [comment]
It was said that there are some datapacks that are known to have this issue. Do we know what the conflict is in those cases? Looking for a repeat of the same problem might be the easiest way to figure it out. I've never done plugin development, but from looking at the error log nothing immediately jumps out at me. The pack doesn't make any changes to bees but it does affect beehives. It's possible this is an issue when a bee attempts to leave or re-enter a beehive. Removing beehives and bee nests from jungle logs might fix it. I'm only speculating.

**Comment by EchocallingEarth** [comment]
If you wanted to be sure if in fact, the Beehive thing could be affecting it. and were willing to upload a copy of the pack less this feature, here, I would be very willing to test. Sadly as I know very little about how plugins or datapacks are created or work exactly all I can do is test and report findings. But I trust those who make datapacks and plugins and know they understand the inner workings of these things. So when asked to test anything I will do so gladly.

**Comment by galdevear** [comment]
I've made a branch on GitHub with the change so you can test it.
https://github.com/galdevear/galds-little-details/tree/slimefun-testing

**Comment by EchocallingEarth** [comment]
> I've made a branch on GitHub with the change so you can test it. https://github.com/galdevear/galds-little-details/tree/slimefun-testing

Will be running full test starting 12-31-22 and report finding back.

Please let me know if there are any particular changes, I should look at in addition to watching for the original issues. Thanks

**Comment by Boomer-1** [comment]
once any code is done and fully tested by following the guidelines on the sf github, you can submit a PR for review at that time.

**Comment by EchocallingEarth** [comment]
Well, I am bummed to report that after testing the GitHub version as provided the error returns. Currently only removing the datapack resolves the issue. Further testing confirmed this issue. To reproduce in stall datapack and restart the server.
